### PR TITLE
Setup state restoration

### PR DESCRIPTION
**Pull request by vladcipariu91**
## Description

...

## Check list

- [ ] I have updated `CHANGELOG.md` if required.
- [ ] I have updated documentation if required.

### CODE REVIEW

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/InteractionModel.kt [hunk 1]
```diff
@@ -57,6 +57,8 @@ open class InteractionModel<InteractionTarget : Any, ModelState : Any>(
         backPressStrategy.init(this, model)
     }
 
+    fun transitionModel() = model
```
Comment (KovalevAndrey) [inline appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/InteractionModel.kt]: let's make `InteractionModel` implement `SavedInstanceStatePlugin`  instead?

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/model/transition/TransitionModel.kt [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 import com.bumble.appyx.interactions.core.Element
 import kotlinx.coroutines.flow.StateFlow
 
-interface TransitionModel<InteractionTarget, ModelState> {
+interface TransitionModel<InteractionTarget, ModelState> : SavesInstanceState {
 
     val output: StateFlow<Output<ModelState>>
 
-    sealed class Output<ModelState> {
+    sealed class Output<ModelState> : java.io.Serializable {
```
Comment (KovalevAndrey) [inline appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/model/transition/TransitionModel.kt]: import?

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/transitionmodel/backstack/BackStackModel.kt [hunk 3]
```diff
@@ -19,7 +19,7 @@ class BackStackModel<InteractionTarget : Any>(
 //    backPressHandler = backPressHandler,
 //    screenResolver = screenResolver,
 //    operationStrategy = operationStrategy,
-//    savedStateMap = savedStateMap,
+    savedStateMap = savedStateMap,
```
Comment (KovalevAndrey) [inline appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/transitionmodel/backstack/BackStackModel.kt:16]: let's add save/restore 2 tests for core models: backstack and spotlight

File: appyx-navigation/src/main/kotlin/com/bumble/appyx/navigation/node/Node.kt [hunk 4]
```diff
@@ -157,7 +157,7 @@
                 if (!plugin.isCorrect()) {
                     Appyx.reportException(
                         IllegalStateException(
-                            "Plugin $plugin has implementation for both BackPressHandler properties, implement only one"
+                            "Plugins.kt $plugin has implementation for both BackPressHandler properties, implement only one"
```
[the remark this answers is not in the document]
Comment (vladcipariu91) [reply appyx-navigation/src/main/kotlin/com/bumble/appyx/navigation/node/Node.kt]: fixed

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/InteractionModel.kt (hunk 1 above)
Comment (vladcipariu91) [reply appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/InteractionModel.kt]: done

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/model/transition/TransitionModel.kt (hunk 2 above)
Comment (vladcipariu91) [reply appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/model/transition/TransitionModel.kt]: done

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/transitionmodel/backstack/BackStackModel.kt (hunk 3 above)
Comment (vladcipariu91) [reply appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/transitionmodel/backstack/BackStackModel.kt:16]: done

**Review by zsoltk — APPROVED** [review]

**Review by KovalevAndrey — APPROVED** [review]

**Pull request merged**